File: TastyAppBackend/src/main/java/com/kondziu/projects/TastyAppBackend/payload/ResetPasswordPayload.java
package com.kondziu.projects.TastyAppBackend.payload;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@Getter
@Setter
public class ResetPasswordPayload {

    private String usernameOrEmail;
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private String password;
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private String token;
    private String frontendUrl;
}

File: TastyApp/src/app/models/ResetPasswordPayload.ts


export class ResetPaswordPayload{
    usernameOrEmail?: string;
    password?: string;
    token?: string;
    frontendUrl?: string;

    constructor(usernameOrEmail: string,
        password: string,
        token: string,
        frontendUrl: string){
            if(usernameOrEmail) this.usernameOrEmail=usernameOrEmail;
            if(password) this.password=password;
            if(token) this.token=token;
            if(frontendUrl) this.frontendUrl=frontendUrl;
        }
}